### PR TITLE
libc: modify open,close... to _NX_OPEN,_NX_CLOSE

### PR DESCRIPTION
**Pull request by W-M-R**
## Summary
libc: modify open,close... to _NX_OPEN,_NX_CLOSE
## Impact
None
## Testing
None

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@W-M-R Please fix:
```
Error: /home/runner/work/nuttx/nuttx/nuttx/libs/libc/lzf/lzf_d.c:5:80: error: Long line found
```

**Comment by acassis** [comment]
@W-M-R I don't remember why these changes were made. Is it necessary to separate kernel from user space APIs?

File: libs/libc/dirent/lib_fdopendir.c [hunk 1]
```diff
@@ -77,14 +79,14 @@ FAR DIR *fdopendir(int fd)
 
   if (!S_ISDIR(st.st_mode))
     {
-      set_errno(ENOTDIR);
+      _NX_SETERRNO(ENOTDIR);
```
Comment (pkarashchenko) [inline libs/libc/dirent/lib_fdopendir.c:82]: seems like we also need change `fstat` to `_NX_STAT` at line 74

File: libs/libc/dirent/lib_nftw.c [hunk 2]
```diff
@@ -198,7 +200,7 @@ do_nftw(FAR char *path, nftw_cb_t fn, int fdlimit, int flags, int level)
 
           if (strlen(de->d_name) > PATH_MAX - j)
             {
-              set_errno(ENAMETOOLONG);
+              _NX_SETERRNO(ENAMETOOLONG);
```
Comment (pkarashchenko) [inline libs/libc/dirent/lib_nftw.c:203]: seems like we need a similar change at line 113 and 164: `get_errno` -> `_NX_GETERRNO`

**Comment by W-M-R** [comment]
https://github.com/apache/nuttx/pull/14651

> @W-M-R I don't remember why these changes were made. Is it necessary to separate kernel from user space APIs?

Yes, kernel mode and user mode need to use different APIs

**Comment by W-M-R** [comment]
Sorry, I accidentally disabled this patch.

**Comment by W-M-R** [comment]
Sorry, I accidentally disabled this patch.

**Comment by W-M-R** [comment]
The automatic test failed. Please give me some time to check it carefully.

**Comment by xiaoxiang781216** [comment]
need rebase to fix the conflict

**Comment by W-M-R** [comment]
ok, it is not necessary to discuss this change, so this patch will be closed

**Comment by acassis** [comment]
> It is not necessary to discuss this change, so this patch will be closed

Please start the discussion at NuttX mailing list!

**Pull request closed without merging**